File: Backend/IFRS/models/Branch.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace IFRS.models;

[Table("branch")]
public class Branch
{
    [Key]
    [Column("pk_branch_id")]
    public int BranchId { get; set; }

    [Column("oracle_code")]
    [StringLength(100)]
    public string? OracleCode { get; set; }

    [Column("branch_code")]
    [StringLength(100)]
    public string? BranchCode { get; set; }

    [Column("branch_name")]
    [StringLength(255)]
    public string? BranchName { get; set; }

    [Column("lessee")]
    [StringLength(255)]
    public string? Lessee { get; set; }

    [Column("status")]
    [StringLength(50)]
    public string? Status { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation property
    public ICollection<Lease> Leases { get; set; } = new List<Lease>();
}

File: Frontend/src/types/branch.ts
import type { LeaseStatus } from "@/types/lease"
import type { ChangeRequest } from "@/types/change-request"

export type BranchStatus = "Active" | "Inactive"

export type BranchLease = {
    branchId: number
    leaseNo: string
    lessor: string
    startDate: string
    endDate: string
    leasePropertyAddress: string
    status: LeaseStatus
}

export type Branch = {
    branchId: number
    oracleCode: string
    branchCode: string
    branchName: string
    lessee: string
    status: BranchStatus
    createdAt?: string
    updatedAt?: string
    hasPendingChangeRequest?: boolean
    leases?: BranchLease[]
}

export type CreateBranchInput = {
    branchName: string
    oracleCode: string
    branchCode: string
    lessee: string
    status: BranchStatus
}

export type UpdateBranchInput = CreateBranchInput & {
    branchId: number
    requestComments?: string
}

export type UpdateBranchResult = {
    message: string
    data: ChangeRequest
}

export type DeleteBranchResult = {
    message: string
}